### PR TITLE
Feat/sagemaker

### PR DESCRIPTION
**Pull request by zagronitay**

### CODE REVIEW

File: resources/sagemaker_endpoint_configurations.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+				Description: "The AWS Region of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSRegion,
+			},
+			{
+				Name:        "kms_key_id",
+				Description: "Amazon Web Services KMS key ID Amazon SageMaker uses to encrypt data when storing it on the ML storage volume attached to the instance.",
+				Type:        schema.TypeString,
+			},
+			{
```
Comment (roneli) [inline resources/sagemaker_endpoint_configurations.go:41]: missing description

File: resources/sagemaker_training_jobs.go [hunk 2]
```diff
@@ -0,0 +1,904 @@
...
+					},
+					{
+						Name:        "training_image",
+						Description: "The registry path of the Docker image that contains the training algorithm",
+						Type:        schema.TypeString,
+					},
+				},
+				Relations: []*schema.Table{
+					{
+						Name:        "aws_sagemaker_training_job_algorithm_specification_metric_definitions",
```
Comment (roneli) [inline resources/sagemaker_training_jobs.go:240]: maybe make this a json as well?

**Review by roneli — CHANGES_REQUESTED** [review]
Partial review, still need to go over training jobs

File: resources/sagemaker_endpoint_configurations.go (hunk 1 above)
Comment (zagronitay) [reply resources/sagemaker_endpoint_configurations.go:41]: https://github.com/aws/aws-sdk-go-v2/blob/56f090f03626e50ab996bd0e8a80ef13f7ab85fc/service/sagemaker/api_op_DescribeEndpointConfig.go#L71

Missing information on this one

File: resources/sagemaker_training_jobs.go [hunk 3]
```diff
@@ -0,0 +1,888 @@
...
+					},
+					{
+						Name:        "status_details",
+						Description: "Details from the rule evaluation.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+			{
+				Name:        "aws_sagemaker_training_job_final_metric_data_list",
```
Comment (roneli) [inline resources/sagemaker_training_jobs.go:368]: I think this can be a json

File: resources/sagemaker_training_jobs.go [hunk 4]
```diff
@@ -0,0 +1,888 @@
...
+						Description: "A list of one or more attribute names to use that are found in a specified augmented manifest file.",
+						Type:        schema.TypeStringArray,
+						Resolver:    schema.PathResolver("DataSource.S3DataSource.AttributeNames"),
+					},
+					{
+						Name:        "data_source_s3_data_source_s3_data_distribution_type",
+						Description: "If you want Amazon SageMaker to replicate the entire dataset on each ML compute instance that is launched for model training, specify FullyReplicated",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("DataSource.S3DataSource.S3DataDistributionType"),
+					},
```
Comment (roneli) [inline resources/sagemaker_training_jobs.go:458]: the name here seems a bit redundant, for example data_source_s3_data_source_s3_data...  -> s3_data_source_distribution_type

File: resources/sagemaker_training_jobs.go [hunk 5]
```diff
@@ -0,0 +1,888 @@
...
+					{
+						Name:        "status_details",
+						Description: "Details from the rule evaluation.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+			{
+				Name:        "aws_sagemaker_training_job_secondary_status_transitions",
```
Comment (roneli) [inline resources/sagemaker_training_jobs.go:575]: this can also be a json imo

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, final comments on the training_jobs resource

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**